app/lib/foreman_deployments/validation/validator.rb

```diff
@@ -0,0 +1,20 @@
+module ForemanDeployments
+  module Validation
+    class Validator
+      def validate(stack_definition)
+        cloned = stack_definition.deep_clone
+
+        validation_visitor = ForemanDeployments::Validation::ValidationVisitor.new
```

Comment:
I'll move the instanciation just before the usage (`cloned.accept(validation_visitor)`)